tests/app/data_model/test_questionnaire_store.py

```diff
@@ -1,185 +1,124 @@
-        store.set_metadata(expected["METADATA"])
-        store.answer_store = AnswerStore(expected["ANSWERS"])
-        store.response_metadata = expected["RESPONSE_METADATA"]
-        store.progress_store = ProgressStore(expected["PROGRESS"])
-
-        # When
-        store.save()  # See setUp - populates self.output_data
-
-        # Then
-        self.assertEqual(expected, json_loads(self.output_data))
-
-    def test_questionnaire_store_errors_on_invalid_object(self):
-        # Given
-        class NotSerializable:
-            pass
-
-        non_serializable_metadata = {"test": NotSerializable()}
-
-        expected = get_basic_input()
-        store = QuestionnaireStore(self.storage)
-        store.set_metadata(non_serializable_metadata)
-        store.response_metadata = expected["RESPONSE_METADATA"]
-        store.answer_store = AnswerStore(expected["ANSWERS"])
-        store.progress_store = ProgressStore(expected["PROGRESS"])
-
-        # When / Then
-        self.assertRaises(TypeError, store.save)
-
-    def test_questionnaire_store_deletes(self):
-        # Given
-        expected = get_basic_input()
-        store = QuestionnaireStore(self.storage)
-        store.set_metadata(expected["METADATA"])
-        store.response_metadata = expected["RESPONSE_METADATA"]
-        store.answer_store = AnswerStore(expected["ANSWERS"])
-        store.progress_store = ProgressStore(expected["PROGRESS"])
-
-        # When
-        store.delete()  # See setUp - populates self.output_data
-
-        # Then
-        self.assertNotIn("a-test-section", store.progress_store)
-        self.assertEqual(store.metadata.copy(), {})
-        self.assertEqual(len(store.answer_store), 0)
-        self.assertEqual(store.response_metadata, {})
-
-    def test_questionnaire_store_raises_when_writing_to_metadata(self):
-        store = QuestionnaireStore(self.storage)
-
-        with self.assertRaises(TypeError):
-            store.metadata["no"] = "writing"
+def test_questionnaire_store_json_loads(questionnaire_store, basic_input):
+    # Given
+    questionnaire_store.input_data = json_dumps(basic_input)
+    # When
+    store = QuestionnaireStore(questionnaire_store.storage)
+    # Then
+    assert store.metadata.copy() == basic_input["METADATA"]
+    assert store.response_metadata == basic_input["RESPONSE_METADATA"]
+    assert store.answer_store == AnswerStore(basic_input["ANSWERS"])
+
+    expected_completed_block_ids = basic_input["PROGRESS"][0]["block_ids"][0]
+
+    assert (
+        len(store.progress_store.get_completed_block_ids("a-test-section", "abc123"))
+        == 1
+    )
+    assert (
+        store.progress_store.get_completed_block_ids("a-test-section", "abc123")[0]
+        == expected_completed_block_ids
+    )
+
+
+def test_questionnaire_store_ignores_extra_json(questionnaire_store, basic_input):
+    # Given
+    basic_input["NOT_A_LEGAL_TOP_LEVEL_KEY"] = "woop_woop_thats_the_sound_of_the_police"
```

Comment:
I'm only migrating tests, not refactoring/checking them.

Reply:
I agree in the main, but obvious issues should be fixed or raised in another story otherwise they get left.

Reply:
Well as far as I can tell when this was introduced, there was no way then for it to even be evaluated, nor is there now. The `basic_input`/jsonified `basic_input` objects aren't passed in to anything that can or ever could process it so as far as I can tell it is useless. The question is, _should_ it have been a feature for metadata to be screened for invalid keys?

Reply:
This was just checking that additional keys aren't made use of by the store. Don't think we really need it. But could add something like `assert not hasattr(store, "not_a_legal_top_level_key")` to give it more meaning but either way it would pass as it is testing, it has no impact essentially.